projects/igniteui-angular/src/lib/grids/filtering/base/grid-filtering-row.component.html

```diff
@@ -27,7 +27,7 @@
             autocomplete="off"
             [value]="value"
             (input)="onInput($event)"
-            [type]="type"
+            type="text"
```

Comment:
This is a really major change in the filtering UI which may cause other problems. The number input actually allows typing - & + signs, so the problem seems to be in the filtering logic itself. I believe the problem is in the IgxGridFilteringRowComponent's "onInput event handler. When the target's value is just "-" (or + actually) it is resolved as NaN, hence the behavior you are observing. I sugges we account for this specific scenario by adding something similar to:
public onInput(eventArgs) {
.....
        const target = eventArgs.target;
        if (!target.value && this.column.dataType === GridColumnDataType.Number) {
            return;
        }
        if (this.column.dataType === GridColumnDataType.DateTime) {
          ....